File: app/src/application/models/Shop.ts
import Item from "./Item";
import Player from "./Player";

export default class Shop
{
    id: string;
    name: string;
    owner: Player;
    items: Item[];
    position: 
    { 
        x: number; 
        y: number; 
    }

    constructor(
        id: string,
        name: string,
        owner: Player,
        items: Item[],
        position: { x: number, y: number }
    ) 
    {
        this.id = id;
        this.name = name;
        this.owner = owner;
        this.position = position;
        this.items = items;
    }
}

File: server/Application/Models/Shop.cs
namespace Ubss.Api.Application.Models;

public sealed class Shop
{
    public Guid Id { get; }
        = Guid.NewGuid();

    public DateTime StoredAt { get; }
        = DateTime.UtcNow;

    public Position Position { get; set; }

    public string Name { get; set; }

    public Player Owner { get; set; }

    public IEnumerable<Item> Items { get; set; }
}
